### PR TITLE
Add SchemaIgnoreClasses property

### LINKED ISSUE #8195: Schema exclude annotation
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes
| RFC         | yes
| BC Break    | no

When using migrations or schema operations, there are facilities to ignore tables via schema_filter setting, see [Ignoring Custom Tables](https://www.doctrine-project.org/projects/doctrine-migrations/en/2.2/reference/generating-migrations.html#ignoring-custom-tables).

However there are many circumstances where entities should not be included for consideration in the schema. For example entities based on views, multiple entities for the same table or view for different sections of a project or serialisation requirements. This list can be quite long. Relying on excluding items just by the table name does not work for a lot of real world situations.

Currently there is no way to prevent a normal entity from being included in the schema list, this can cause schema failures and the inability to use the migration or schema tools in Doctrine.

I would like to propose adding a @SchemaExclude annotation which could be tested for in the SchemaTool::procesingNotRequired method and prevent the entity table from being added to the schema table list.

Many thanks.
Ian.

### PR DESCRIPTION
**Pull request by simPod**
Supersedes #8203
Fixes #8195

### CODE REVIEW

**Comment by simPod** [comment]
@greg0ire rebased, diff looks the same, CI is green. I guess it's k

**Comment by greg0ire** [comment]
@simPod doesn't this require any documentation? Also, is it really a bugfix? Looks more like a new feature to me, in which case you should retarget to 2.11.x

**Comment by simPod** [comment]
It was already a feature in #8203 😄 I'll rebase and retarget then.

**Review by SenseException — CHANGES_REQUESTED** [review]
https://www.doctrine-project.org/projects/doctrine-orm/en/2.11/reference/advanced-configuration.html would need the additional config.

File: docs/en/reference/advanced-configuration.rst [hunk 1]
```diff
@@ -440,6 +440,21 @@ That will be available for all entities without a custom repository class.
...
+SchemaTool will then skip these (e.g. when comparing schemas).
+
+.. code-block:: php
+
+    <?php
+    $config->setSchemaIgnoreClasses([$fqcn]);
+    $config->getSchemaIgnoreClasses();
+
+The default value is ``Doctrine\ORM\EntityRepository``.
+Any repository class must be a subclass of EntityRepository otherwise you got an ORMException
```
Comment (SenseException) [inline docs/en/reference/advanced-configuration.rst]: I'm confused with this part of the explanation. It's talking about fqcn of entities, but here it references repositories.

Comment (simPod) [reply docs/en/reference/advanced-configuration.rst]: Wrong copy paste, forgot to remove these lines. Fixed, thx

**Review by SenseException — APPROVED** [review]

**Review by beberlei — APPROVED** [review]

**Comment by beberlei** [comment]
@simPod thanks!

**Pull request merged**